pkg/proxmox/goproxmox/api_client.go

```diff
@@ -142,10 +142,14 @@ func (c *APIClient) FindVMResource(ctx context.Context, vmID uint64) (*proxmox.C
 	return nil, fmt.Errorf("unable to find VM with ID %d on any of the nodes", vmID)
 }
 
-// FindVMTemplateByTags tries to find a VMID by its tags across the whole cluster.
-func (c *APIClient) FindVMTemplateByTags(ctx context.Context, templateTags []string) (string, int32, error) {
-	vmTemplates := make([]*proxmox.ClusterResource, 0)
+// FindVMTemplatesByTags finds VM templates by tags across the whole cluster.
+func (c *APIClient) FindVMTemplatesByTags(ctx context.Context, templateTags []string, allowedNodes []string, localStorage bool) (map[string]int32, error) {
```

Comment:
Again, since you can't see the sonarqube report: It finds this function too complex ("Refactor this method to reduce its Cognitive Complexity from 21 to the 15 allowed"). While indeed there are many branches here, they're all fairly straightforward and there isn't really a better place to put them.